pkg/scheduler/scheduler.go

```diff
@@ -121,9 +144,19 @@ func (s *Scheduler) schedule(ctx context.Context) {
 			e.inadmissibleMsg = "cohort used in this cycle"
 			continue
 		}
+
+		if s.waitForPodsReady {
+			// Block admission until all currently admitted workloads are in
+			// PodsReady condition if the waitForPodsReady is enabled
+			s.cache.WaitForPodsReady(ctx)
+		}
+
 		log := log.WithValues("workload", klog.KObj(e.Obj), "clusterQueue", klog.KRef("", e.ClusterQueue))
 		if err := s.admit(ctrl.LoggerInto(ctx, log), e); err != nil {
 			e.inadmissibleMsg = fmt.Sprintf("Failed to admit workload: %v", err)
+		} else if s.waitForPodsReady {
+			klog.InfoS("Workload successfully admitted when waitForPodsReady - do admit more in this cycle")
```

Comment:
do not?

Reply:
Yes, but I think this if is no longer necessary once we moved the waiting inside the loop over heads.